ecs-init/docker/docker.go

```diff
@@ -427,6 +432,65 @@ func getDockerPluginDirBinds() []string {
 	return pluginBinds
 }
 
+// take pathPredicate as an argument for unit testing
+func getCapabilityExecBinds(pathPredicate func(path string, predicate func(fileInfo os.FileInfo) bool) (bool, error)) []string {
+	const (
+		capabilityExecName = "exec"
+
+		hostCapabilitiesResourcesRootDir      = "/home/ec2-user/capabilities-deps"
+		containerCapabilitiesResourcesRootDir = "/capabilities"
+
+		hostBinRelativePath = "bin"
+		hostCertsDir        = "/etc/pki/ca-trust/extracted/pem"
+		hostCertsFilename   = "tls-ca-bundle.pem"
+
+		containerBinRelativePath   = "bin"
+		containerCertsRelativePath = "certs"
+
+		// SSM-related settings are not configurable for now
+		// containerCapabilityExecConfigPath = "config"
+	)
+
+	isDir := func(fileInfo os.FileInfo) bool {
+		return fileInfo.IsDir()
+	}
+	isFile := func(fileInfo os.FileInfo) bool {
+		return !fileInfo.IsDir()
+	}
+	hostCapabilityExecResourcesDir := filepath.Join(hostCapabilitiesResourcesRootDir, capabilityExecName)
+	containerCapabilityExecResourcesDir := filepath.Join(containerCapabilitiesResourcesRootDir, capabilityExecName)
+
+	binds := []string{}
+
+	// bind mount the entire /deps/exec/bin folder for higher flexibility
+	// minimal change required to add other ssm binaries as dependency in the future (just need to be placed inside the bin directory)
+	hostCapabilityExecBinDir := filepath.Join(hostCapabilityExecResourcesDir, hostBinRelativePath)
+	if exists, err := pathPredicate(hostCapabilityExecBinDir, isDir); err == nil && exists {
+		binds = append(binds, hostCapabilityExecBinDir+":"+filepath.Join(containerCapabilityExecResourcesDir, containerBinRelativePath)+readOnly)
+	}
+
+	// bind mount this specific cert file for now, CertsDir and CertsFile might be changed to be configurable in the future
+	hostCertsFile := filepath.Join(hostCertsDir, hostCertsFilename)
+	if exists, err := pathPredicate(hostCertsFile, isFile); err == nil && exists {
+		binds = append(binds, hostCertsFile+":"+filepath.Join(containerCapabilityExecResourcesDir, containerCertsRelativePath, hostCertsFilename)+readOnly)
+	}
+
+	return binds
+}
+
+func pathExistsAndValid(path string, predicate func(fileInfo os.FileInfo) bool) (bool, error) {
```

Comment:
could you add unit tests for this method?

Reply:
I would argue that there is not much value doing that since this is a very thin wrapper on top of `os.Stat`, adding UTs for this would be almost identical to testing golang's `os` library

Reply:
+1 on adding test, it's easy to return the wrong thing, like returning `false` when it should have been `true`. If someone changes this code in the future and returns the wrong thing, then exec might break and we would realize late in the process, better to catch things early

Reply:
got it, thanks for the explanation! will add unit test to verify that we're using the expected api